src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -1007,9 +1007,24 @@ public BoundExpression Convert(TypeSymbol type, BoundExpression arg, ConversionK
             { WasCompilerGenerated = true };
         }
 
-        public BoundExpression Array(TypeSymbol elementType, BoundExpression[] elements)
+        public BoundExpression ArrayOrEmpty(TypeSymbol elementType, BoundExpression[] elements)
```

Comment:
We should add a doc comment to ArrayOrEmpty methods making it clear that using them should be a conscious choice regarding usage of Array.Empty in that particular scenario. Perhaps the name should be changed to something better reflecting the outcome, "ArrayOrEmptySingleton" for example.